core/src/main/scala/org/scalaexercises/algebra/progress.scala

```diff
@@ -10,55 +10,43 @@ import org.scalaexercises.types.user._
 import org.scalaexercises.types.exercises._
 import org.scalaexercises.types.progress._
 
-import cats.Monad
-import cats.implicits._
+import cats.{ Monad, Applicative, Unapply }
 import cats.free._
+import cats.implicits._
 
-/** Users Progress Ops GADT
-  */
-sealed trait UserProgressOp[A]
-
-final case class GetLastSeenSection(
-  user:    User,
-  library: String
-) extends UserProgressOp[Option[String]]
+import freestyle._
+import freestyle.implicits._
 
-final case class GetExerciseEvaluations(
-  user:    User,
-  library: String,
-  section: String
-) extends UserProgressOp[List[UserProgress]]
+object imp {
```

Comment:
Not sure why this is necessary. The `Free` companion in cats already includes a monad instance that should satisfy the Applicative requirement